packages/core/test/acceptance/di_spec.ts

```diff
@@ -1919,6 +1919,128 @@ describe('di', () => {
              });
         });
       });
+
+      onlyInIvy('Ivy support InjectFlags in Injector')
+          .it('should support Optional flag in Injector', () => {
+            @Directive({selector: '[dirA]'})
+            class DirectiveA {
+              dirB = this.injector.get(DirectiveB, null, InjectFlags.Optional);
+              constructor(public injector: Injector) {}
+            }
+            @Component({template: '<div dirA></div>'})
+            class MyComp {
+              @ViewChild(DirectiveA) dirA!: DirectiveA;
+            }
+
+            TestBed.configureTestingModule({declarations: [DirectiveA, DirectiveB, MyComp]});
+            const fixture = TestBed.createComponent(MyComp);
+            fixture.detectChanges();
+
+            const dirA = fixture.componentInstance.dirA;
+            expect(dirA.dirB).toBeNull();
+          });
```

Comment:
I think you can simplify this test to have just one component:

```suggestion
          .it('should support Optional flag in Injector', () => {
            const NON_EXISTING_PROVIDER = new InjectionToken('non-existing');
    
            @Component({template: '...'})
            class MyComp {
              provider = null;
              constructor(public injector: Injector) {
                this.provider = this.injector.get(NON_EXISTING_PROVIDER, null, InjectFlags.Optional);
              }
            }

            TestBed.configureTestingModule({declarations: [MyComp]});
            const fixture = TestBed.createComponent(MyComp);
            fixture.detectChanges();

            expect(fixture.componentInstance.provider).toBeNull();
          });
```